lib/galaxy/model/__init__.py

```diff
@@ -8921,12 +8921,13 @@ def to_dict(self, view="collection", value_mapper=None, step_details=False, lega
 
             input_parameters = {}
             for input_step_parameter in self.input_step_parameters:
-                label = input_step_parameter.workflow_step.label
-                if not label:
-                    continue
-                input_parameters[label] = {
+                if security:
+                    step_id = security.encode_id(input_step_parameter.workflow_step_id)
+                else:
+                    step_id = input_step_parameter.workflow_step_id
+                input_parameters[step_id] = {
```

Comment:
That's a breaking change, we can't do that

Reply:
Ok, but we still need to include input parameters without labels in the `invocation.input_parameters`, so do we just use dummy labels instead? Do we pass a dummy label from the frontend or do we set one in the backend in this same function?